src/libraries/Common/src/System/Net/Security/NetEventSource.Security.Windows.cs

```diff
@@ -1,13 +1,16 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics.CodeAnalysis;
 using System.Diagnostics.Tracing;
 using System.Net.Security;
 
 namespace System.Net
 {
     internal sealed partial class NetEventSource
     {
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+            Justification = "parameter intent is an enum and is trimmer safe")]
```

Comment:
Why is it that we pass the `intent` as enum here, but we cast it to `int` in other events? Maybe we should convert to `int` always?

Reply:
You are right, I see this pattern to be inconsistent. For example, HttpTelemetry has an Event Write that takes an enum. I'd ideally like to treat this as a separate issue since both are trimmer-safe. I opened an issue to add a test to monitor these event signatures